### PR TITLE
generic pagination + getting streetcodes on main page with pagination/front

### PR DESCRIPTION
**Pull request by malyna2**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by Oleh-Kril — CHANGES_REQUESTED** [review]
Weird behaviour with requests (I changed pageSize to 1 when was testing)
Screenshot 2023-10-12 at 10 04 52 PM
/assets/91199925/c8e02e31-3dac-4d4e-8732-2329b56f0671">
Screenshot 2023-10-12 at 10 06 24 PM

File: src/features/MainPage/StreetcodeSlider/StreetcodeSlider.component.tsx [hunk 1]
```diff
@@ -37,19 +40,24 @@ const StreetcodeSlider = () => {
...
-                    });
+        const newStreetcodes: StreetcodeMainPage[] = [];
+        const newImages: Image[] = [];
+        while (true) {
+            try {
+                const response = await fetchNextPageOfStreetcodesMainPage();
+                newStreetcodes.push(...response);
+                setStreetcodes(newStreetcodes);
+                for (const streetcode of response) {
+                    await ImagesApi.getById(streetcode.imageId)
```
[the remark this answers is not in the document]
Comment (malyna2) [reply src/features/MainPage/StreetcodeSlider/StreetcodeSlider.component.tsx]: Solved this problem. await was needed to make sure we recieved all images before setting to useState

**Comment by malyna2** [comment]
Issue with loading of all streetcodes can be reproduced only with strict mode

File: src/features/MainPage/StreetcodeSlider/StreetcodeSlider.component.tsx [hunk 2]
```diff
@@ -37,17 +40,34 @@ const StreetcodeSlider = () => {
...
+                try {
+                    const newStreetcodes = await fetchNextPage();
+                    setStreetcodes((prevState) => [...prevState, ...newStreetcodes]);
+
+                    const newImages: Image[] = [];
+                    const pomises = [];
+
+                    for (let i = 0; i < newStreetcodes.length; i++) {
+                        const currentPosition = newImages.length + i;
+                        pomises.push(ImagesApi.getById(newStreetcodes[i].imageId).then((img) => {
```
Comment (Oleh-Kril) [inline src/features/MainPage/StreetcodeSlider/StreetcodeSlider.component.tsx:58]: typo

**Review by Oleh-Kril — APPROVED** [review]
Good job

**Pull request merged**